lib/actions/apiV2.js

```diff
@@ -601,6 +616,7 @@ export const findRoute = (params) =>
             })
             newRoute.patterns = routePatterns
             // TODO: avoid explicit behavior shift like this
+            newRoute.agency.id = getAgencyIdMatchingConfig(newRoute.agency)
```

Comment:
This is a clever solution to this problem, but there is a problem: the transit operator agency id is also used to display operator logos in the itinerary body and stop viewer. The agency ids need to be of the same format there! Either that or the config needs to support defining multiple agency ids to match against

Reply:
Okay, so it sounds like the full agency id returned from OTP should be used and my assumptions about nomenclature were incorrect.